### DISCUSSION TITLE
The future of PnP Modern Search -- What are the plan for the future ?

### DISCUSSION THREAD

**Discussion by ferrarirosso** [microsoft-search/pnp-modern-search#3290] [Ideas]
The PnP Modern Search is a great solution with a lot of features and gives a lot of benefits for customers.
A huge thanks to all contributors and maintainers !

Release 4.9.3 is out, there are still a couple of numbers until 5.x, but I just want to have some visibility if there are some plans for a release 5.x and if there are such plans, is this going to include some new "killer" features, like getting data for example from the Semantic Index or from Azure Cognitive Service, ....

Thanks !

**Comment by wobba** [comment]
We'll keep on being on v4 until there is some huge internal architectural change in the web parts which breaks upgrades.

ACS can be supported today via a custom search provider (extensibility) and semantic results will be an increment when API's add support for it.

**Reply by ferrarirosso** [reply]
@wobba thanks for pointing me to the custom search provider (extensiblity)
Did you man like adding a new datasource here ?
https://github.com/microsoft-search/pnp-modern-search/tree/main/search-parts/src/dataSources
The only technical issue that I have is mapping permissions or making the ACS content SPO permissions aware.
Are there any APIs / examples to implement / react to permission-changes in SPO ?

**Reply by wobba** [reply]
ACL's in SharePoint are a PITA to translate to AAD from my very old experience when I created a SP crawler way back. And I have next to no ACS experience.

**Reply by ferrarirosso** [reply]
Thanks, i can imagine. Accessing ACLs is doable but reacting to ACL changes is imho not possible, am I right ?

**Comment by FranckyC** [comment]
Hi @ferrarirosso, I can give you some hints regarding the future of this solution. On my side, I didn't invest so much time this year to the current v4 solution and do not plan to do so for an upcoming v5 version based on this architecture (btw, you're not forced to switch to 5.x when you reach 4.10 :D). I think the current version is good and stable enough to already cover many common scenarios. Investing time in new features won't give much value at this point in my opinion.

However, what I can tell is the PnP Modern Search initiative will evolve soon to a new approach, focused on Microsoft Search only. I think it is time to move away from tradionnal SharePoint API and use new Microsoft Search capabilities to cover a broader search experience. For instance, regarding Azure Cognitive Services integration, this is typically something Microsoft Search can do with Graph connectors so my recommendations would be to avoid creating a custom data source for this.

Stay tuned, accouncements should come early december :D

**Reply by ferrarirosso** [reply]
Thanks.
SharePoint offers a lot of customisations when it comes to search. If we can map those possibilities to MS Search, that's ok for me :-)
I was more thinking about implementing a RAG based on [this](https://docs.llamaindex.ai/en/stable/end_to_end_tutorials/dev_practices/production_rag.html) article

Let's see what comes early december :-)
